js/ace.js

```diff
@@ -0,0 +1,1021 @@
+                        'max': this.safeNumber (market, 'maxLimitBaseAmount'),
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'precision': {
+                    'price': this.safeNumber (market, 'basePrecision'),
+                    'amount': this.safeNumber (market, 'quotePrecision'),
+                },
+                'active': undefined,
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //
+        //     {
+        //         "base_volume":229196.34035399999,
+        //         "last_price":11881.06,
+        //         "quote_volume":19.2909
+        //     }
+        //
+        const marketId = this.safeString (ticker, 'id');
+        market = this.safeMarket (marketId);
+        const symbol = this.safeString (market, 'symbol');
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'high': undefined,
+            'low': undefined,
+            'bid': undefined,
+            'bidVolume': undefined,
+            'ask': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': this.safeString (ticker, 'last_price'),
+            'last': this.safeString (ticker, 'last_price'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeString (ticker, 'base_volume'),
+            'quoteVolume': this.safeString (ticker, 'quote_volume'),
+            'info': ticker,
+        }, market);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name ace#fetchTicker
+         * @description fetches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const response = await this.publicGetOapiV2ListTradePrice (params);
+        const marketId = market['id'];
```

Comment:
we should only add artificial fields as a last resort, here I don't think we really need to do it, since you have the market you can just do this inside `parseTicker` 

```Javascript
const symbol = this.safeSymbol (undefined, market);
```